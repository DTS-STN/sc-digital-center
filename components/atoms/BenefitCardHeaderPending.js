import en from '../../locales/en'
import fr from '../../locales/fr'
import HorizontalRule from './HorizontalRule'
import BenefitCode from '../../constants/BenefitCode'

export default function BenefitCardHeaderPending(props) {
  const t = props.locale === 'en' ? en : fr
  const getBenefitCardTitle = () => {
    if (props.benefit.benefitType === BenefitCode.cppd) {
      return t[BenefitCode.cpp.toLowerCase()]
    } else {
      return t[props.benefit.benefitType.toLowerCase()]
    }
  }

  return (
    <div className="px-4 md:px-6">
      <div className="flex mr-12 justify-end">
        <h2
          className={
            'font-bold font-display text-white text-lg px-5 py-1 ml-8 mb-5 rounded-b-md bg-gray-pending'
          }
        >
          {props.benefit.applicationStatus}
        </h2>
      </div>
      <div className="mx-auto sm:grid sm:grid-cols-4 sm:divide-x-2">
        <div
          id={`${props.benefit.benefitType}-pending`}
          className="col-span-1 py-4 md:px-0 lg:px-3"
        >
          <div className="font-bold font-display text-4xl sm:text-2xl lg:text-4xl mb-2 w-44 sm:w-32 lg:w-44">
            {getBenefitCardTitle()}
          </div>
        </div>
        <HorizontalRule width="w-1/3" visibility="sm:hidden" />
        <div
          id={`${props.benefit.benefitType}-pending-paymentStartDate`}
          className="grid col-span-3 gap-y-4 gap-x-1 sm:grid-cols-3 font-display sm:pl-8 lg:pl-10"
        >
          <div id={`${props.benefit.benefitType}-pending-applicationDate`}>
            <p className="text-base sm:pb-2">{t.applicationSubmitted}</p>
            <p className="font-bold text-lg">{props.benefit.applicationDate}</p>
          </div>

          <div
            id={`${props.benefit.benefitType}-pending-estimatedDateOfDecision`}
          >
            <p className="text-base sm:pb-2">{t.estimatedDateOfDecision}</p>
            <p className="font-bold text-lg">
              {props.benefit.estimatedDateOfDecision}
            </p>
          </div>

          <div id={`${props.benefit.benefitType}-pending-progressbar`}>
            <p className="font-display sm:pb-2">{t.latestStatus}</p>
            <p className="font-display font-bold text-lg">
              {props.benefit.benefitStatusProgress}
            </p>
            <p className="font-bold text-lg">
              {props.benefit.latestStatusDate}
            </p>
            <a
              href={t.url_statusAndMessages}
              className="mt-1 text-bright-blue-solid underline"
            >
              {t.viewMyStatusAndMessages}
            </a>
          </div>
        </div>
      </div>
    </div>
  )
}
