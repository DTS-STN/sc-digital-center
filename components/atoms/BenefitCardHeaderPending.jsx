import en from '../../locales/en'
import fr from '../../locales/fr'

export default function PendingHeader(props) {
  const t = props.locale === 'en' ? en : fr

  return (
    <div className="layout-container">
      <div className="flex mr-12 justify-end">
        <h2
          className={
            'font-bold font-display text-lg ml-2 px-5 py-1 ml-8 mb-5 rounded-b-md bg-orange-400'
          }
        >
          {props.benefit.applicationStatus}
        </h2>
      </div>

      <div className="sm:flex sm:justify-around sm:grid sm:grid-cols-4 divide-x-2 sm:mx-auto">
        <div id="paymentStartDate" className="col-span-1 px-3 py-4">
          <div className="font-bold font-display text-5xl mb-2">
            {props.benefit.benefitType}
          </div>
          <p className="font-display font-bold text-lg">
            {props.benefit.benefitName}
          </p>
        </div>

        <div
          id="paymentStartDate"
          className="font-display col-span-3 sm:grid grid-cols-3"
        >
          <div id="paymentEndDate" className="pl-10 py-4">
            <p className="text-base">{t.applicationSubmitted}</p>
            <p className="font-bold text-lg">{props.benefit.applicationDate}</p>
          </div>

          <div id="paymentEndDate" className="pl-10 py-4">
            <p className="text-base w-44">
              {t.estimatedDateOfReviewCompletion}
            </p>
            <p className="font-bold text-lg">
              {props.benefit.estimatedReviewCompletionDate}
            </p>
          </div>

          {/* Progressbar Placeholder */}
          <div id="paymentEndDate" className="pl-10 py-4">
            <p className="font-display font-bold text-lg">{'Progressbar'}</p>
            <p className="font-display text-base">4 out of 6 review</p>
            <p className="font-display text-base">steps complete</p>
          </div>
        </div>
      </div>
    </div>
  )
}
