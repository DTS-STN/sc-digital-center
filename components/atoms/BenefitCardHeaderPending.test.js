import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import BenefitCardHeaderPending from './BenefitCardHeaderPending'
import { axe, toHaveNoViolations } from 'jest-axe'

const pendingCPP = {
  benefitType: 'CPP',
  benefitName: 'Canada Pension Plan',
  status: 'Pending',
  statusDescription:
    'Your application is pending, we will notify you with decision',
  applicationDate: 'July 1, 2021',
  estimatedDateOfDecision: 'July 31, 2021',
  applicationType: 'Canada Pension Plan (CPP)',
  applicationDescription: 'Online application received',
  nextPaymentAmount: 735.34,
  nextPaymentDate: 'September 30, 2021',
  pensionStartDate: 'August 1, 2021',
  payeeFullName: 'John Smith Doe',
  payeeAddress: '123 - 00 Fake Street, Ottawa, On A1A-1A1',
  accountNumber: 'XXXX-123',
  institutionNumber: '002',
  payerName: 'Government of Canada',
  paymentDepositDate: 'August 30, 2021',
  paymentStatus: 'Issued',
  paymentType: 'Direct Deposit',
  additionalInformation:
    'We will notify you as soon as we have process your application.',

  applicationStatus: 'Application Submitted',
  pendingBenefits: 'Retirement pension',
  benefitStatusProgress: 'In Progress',
}

const pendingCPPD = {
  benefitType: 'CPPD',
  benefitName: 'Canada Pension Plan Disability',
  status: 'Pending',
  statusDescription:
    'Your application is pending, we will notify you with decision',
  applicationDate: 'July 11, 2021',
  estimatedDateOfDecision: 'July 31, 2021',
  applicationType: 'Canada Pension Plan (CPP)',
  applicationDescription: 'Online application received',
  nextPaymentAmount: 735.34,
  nextPaymentDate: 'September 30, 2021',
  pensionStartDate: 'August 1, 2021',
  payeeFullName: 'John Smith Doe',
  payeeAddress: '123 - 00 Fake Street, Ottawa, On A1A-1A1',
  accountNumber: 'XXXX-123',
  institutionNumber: '002',
  payerName: 'Government of Canada',
  paymentDepositDate: 'August 30, 2021',
  paymentStatus: 'Issued',
  paymentType: 'Direct Deposit',
  additionalInformation:
    'We will notify you as soon as we have process your application.',

  applicationStatus: 'Application Submitted',
  pendingBenefits: 'Retirement pension',
  benefitStatusProgress: 'In Progress',
}

describe('BenefitCardHeaderPending Tests', () => {
  it('renders the Header of a Pending Benefit', () => {
    render(<BenefitCardHeaderPending benefit={pendingCPP} locale={'en'} />)
    const titleText = screen.getByText('Canada Pension Plan')
    const applicationStatusProgress = screen.getByText('In Progress')
    const applicationStatus = screen.getByText('Application Submitted')
    expect(titleText).toBeInTheDocument()
    expect(applicationStatus).toBeInTheDocument()
    expect(applicationStatusProgress).toBeInTheDocument()
  })

  it('renders the Header of a CPPD', () => {
    render(<BenefitCardHeaderPending benefit={pendingCPPD} locale={'en'} />)
    const titleTextCPPD = screen.getByText('Canada Pension Plan')
    expect(titleTextCPPD).toBeInTheDocument()
    const titleTextDisability = screen.queryByText(
      'Canada Pension Plan Disability'
    )
    expect(titleTextDisability).toBeNull()
  })

  it('has no a11y violations', async () => {
    const { container } = render(
      <BenefitCardHeaderPending benefit={pendingCPP} locale={'en'} />
    )
    const results = await axe(container)
    expect(results).toHaveNoViolations
  })
})
