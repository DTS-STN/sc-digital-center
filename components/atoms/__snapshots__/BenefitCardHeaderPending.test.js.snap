// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BenefitCardHeaderPending Tests renders the Header of a Pending Benefit 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="layout-container"
      >
        <div
          class="flex mr-12 justify-end"
        >
          <h2
            class="font-bold font-display text-lg px-5 py-1 ml-8 mb-5 rounded-b-md bg-orange-pending"
          />
        </div>
        <div
          class="mx-auto sm:grid sm:grid-cols-4 sm:divide-x-2"
        >
          <div
            class="col-span-1 py-4 md:px-0 lg:px-3"
            id="benefitName"
          >
            <div
              class="font-bold font-display text-4xl sm:text-2xl lg:text-4xl mb-2 w-44 sm:w-32 lg:w-44"
            />
          </div>
          <div
            class="sm:hidden w-1/3 flex-grow border-t border-gray-light"
          />
          <div
            class="font-display col-span-3 sm:grid grid-cols-3"
            id="paymentStartDate"
          >
            <div
              class="py-4 sm:pl-10"
              id="paymentEndDate"
            >
              <p
                class="text-base"
              />
              <p
                class="font-bold text-lg"
              />
            </div>
            <div
              class="py-4 sm:pl-10"
              id="paymentEndDate"
            >
              <p
                class="text-base w-44"
              />
              <p
                class="font-bold text-lg"
              />
            </div>
            <div
              class="py-4 sm:pl-10"
              id="paymentEndDate"
            >
              <p
                class="font-display font-bold text-lg"
              >
                Progressbar
              </p>
              <p
                class="font-display text-base"
              >
                4 out of 6 review
              </p>
              <p
                class="font-display text-base"
              >
                steps complete
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="layout-container"
    >
      <div
        class="flex mr-12 justify-end"
      >
        <h2
          class="font-bold font-display text-lg px-5 py-1 ml-8 mb-5 rounded-b-md bg-orange-pending"
        />
      </div>
      <div
        class="mx-auto sm:grid sm:grid-cols-4 sm:divide-x-2"
      >
        <div
          class="col-span-1 py-4 md:px-0 lg:px-3"
          id="benefitName"
        >
          <div
            class="font-bold font-display text-4xl sm:text-2xl lg:text-4xl mb-2 w-44 sm:w-32 lg:w-44"
          />
        </div>
        <div
          class="sm:hidden w-1/3 flex-grow border-t border-gray-light"
        />
        <div
          class="font-display col-span-3 sm:grid grid-cols-3"
          id="paymentStartDate"
        >
          <div
            class="py-4 sm:pl-10"
            id="paymentEndDate"
          >
            <p
              class="text-base"
            />
            <p
              class="font-bold text-lg"
            />
          </div>
          <div
            class="py-4 sm:pl-10"
            id="paymentEndDate"
          >
            <p
              class="text-base w-44"
            />
            <p
              class="font-bold text-lg"
            />
          </div>
          <div
            class="py-4 sm:pl-10"
            id="paymentEndDate"
          >
            <p
              class="font-display font-bold text-lg"
            >
              Progressbar
            </p>
            <p
              class="font-display text-base"
            >
              4 out of 6 review
            </p>
            <p
              class="font-display text-base"
            >
              steps complete
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
